std/ascii.zig

```diff
@@ -0,0 +1,232 @@
+
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    };
+    const punct = []u1{
+    //  0, 1, 2, 3, 4, 5, 6, 7 ,8, 9,10,11,12,13,14,15
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1,
+
+        1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1,
+        1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0,
+    };
+    const graph = []u1{
+    //  0, 1, 2, 3, 4, 5, 6, 7 ,8, 9,10,11,12,13,14,15
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0,
+    };
+
+    comptime var i = 0;
+    inline while (i < 128) : (i += 1) {
+        table[i] =
+            u8(alpha[i]) << @enumToInt(tIndex.Alpha) |
+            u8(hex[i])   << @enumToInt(tIndex.Hex) |
+            u8(space[i]) << @enumToInt(tIndex.Space) |
+            u8(digit[i]) << @enumToInt(tIndex.Digit) |
+            u8(lower[i]) << @enumToInt(tIndex.Lower) |
+            u8(upper[i]) << @enumToInt(tIndex.Upper) |
+            u8(punct[i]) << @enumToInt(tIndex.Punct) |
+            u8(graph[i]) << @enumToInt(tIndex.Graph);
+    }
+    mem.set(u8, table[128..256], 0);
+    break :init table;
+};
+
+fn inTable(c: u8, t: tIndex) bool {
+    return (combinedTable[c] & (u8(1) << @enumToInt(t))) != 0;
+}
+
+pub fn isAlNum(c: u8) bool {
+    return (combinedTable[c] & ((u8(1) << @enumToInt(tIndex.Alpha)) | 
+                                 u8(1) << @enumToInt(tIndex.Digit))) != 0;
+}
+
+pub fn isAlpha(c: u8) bool {
+    return inTable(c, tIndex.Alpha);
+}
+
+pub fn isCtrl(c: u8) bool {
+    return c < 0x20 or c == 127; //DEL
+}
+
+pub fn isCntrl(c: u8) bool {
+    return isCtrl(c);
+}
+
+pub fn isDigit(c: u8) bool {
+    return inTable(c, tIndex.Digit);
+}
+
+pub fn isGraph(c: u8) bool {
+    return inTable(c, tIndex.Graph);
+}
+
+pub fn isLower(c: u8) bool {
+    return inTable(c, tIndex.Lower);
+}
+
+pub fn isPrint(c: u8) bool {
+    return inTable(c, tIndex.Graph) or c == ' ';
+}
+
+pub fn isPunct(c: u8) bool {
+    return inTable(c, tIndex.Punct);
+}
+
+pub fn isSpace(c: u8) bool {
+    return inTable(c, tIndex.Space);
+}
+
+pub fn isUpper(c: u8) bool {
+    return inTable(c, tIndex.Upper);
+}
+
+pub fn isXDigit(c: u8) bool {
```

Comment:
Do these function names/capitalisation make sense?
`isPunct` is cutting off letters
`isASCII` is mostly caps
`isXDigit` looks weird

Related to #1884 I guess?